### PR TITLE
release 0.6.0

### PR DESCRIPTION
**Pull request by zachschuermann**
shipping CDF and a number of other improvements!

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,18 +1,45 @@
...
+4. Remove lifetime requirement on `Scan::execute` (#588)
+5. `scan::Scan::predicate` renamed as `physical_predicate` to eliminate ambiguity (#512)
+6. `scan::log_replay::scan_action_iter` now takes fewer (and different) params. (#512)
+7. `Expression:Unary`, `Expression::Binary`, and `Expression::Variadic` now wrap a struct of the
+   same name containing their fields (#530)
+8. New `Error` variants: `Error:ChangeDataFeedIncompatibleSchema`
 
 *Additions*
+1. Ability to read a table's change data feed with new TableChanges API! See new `table_changes`
+   module as well as the 'read_table_changes' example. Changes include:
```
Comment (OussamaSaoudi-db) [inline CHANGELOG.md:22]:
```suggestion
   module as well as the 'read-table-changes' example. Changes include:
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,18 +1,65 @@
...
 
 *Additions*
+1. Ability to read a table's change data feed with new TableChanges API! See new `table_changes`
+   module as well as the 'read-table-changes' example. Changes include:
+  - Implement Log Replay for Change Data Feed ([#540])
+  - `ScanFile` expression and visitor for CDF ([#546])
+  - Resolve deletion vectors to find inserted and removed rows for CDF ([#568])
+  - Helper methods for CDF Physical to Logical Transformation ([#579])
+  - `TableChangesScan::execute` and end to end testing for CDF ([#580])
+  - `TableChangesScan::schema` method to get logical schema ([#589])
```
Comment (OussamaSaoudi-db) [inline CHANGELOG.md:28]: May want to link the PR for `read-table-changes`. Besides that, LGTM!

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,18 +1,65 @@
...
 *Breaking*
+1. `Scan::execute` takes an `Arc<dyn EngineData>` now ([#553])
+2. `StructField::physical_name` no longer takes a `ColumnMapping` argument ([#543])
+3. removed `ColumnMappingMode` `Default` implementation ([#562])
+4. Remove lifetime requirement on `Scan::execute` ([#588])
+5. `scan::Scan::predicate` renamed as `physical_predicate` to eliminate ambiguity ([#512])
+6. `scan::log_replay::scan_action_iter` now takes fewer (and different) params. ([#512])
+7. `Expression:Unary`, `Expression::Binary`, and `Expression::Variadic` now wrap a struct of the
+   same name containing their fields ([#530])
+8. New `Error` variant `Error:ChangeDataFeedIncompatibleSchema`
```
Comment (OussamaSaoudi-db) [inline CHANGELOG.md:18]:
```suggestion
8. New `Error` variant `Error::ChangeDataFeedIncompatibleSchema`
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,18 +1,65 @@
...
 **API Changes**
 
 *Breaking*
+1. `Scan::execute` takes an `Arc<dyn EngineData>` now ([#553])
+2. `StructField::physical_name` no longer takes a `ColumnMapping` argument ([#543])
+3. removed `ColumnMappingMode` `Default` implementation ([#562])
+4. Remove lifetime requirement on `Scan::execute` ([#588])
+5. `scan::Scan::predicate` renamed as `physical_predicate` to eliminate ambiguity ([#512])
+6. `scan::log_replay::scan_action_iter` now takes fewer (and different) params. ([#512])
+7. `Expression:Unary`, `Expression::Binary`, and `Expression::Variadic` now wrap a struct of the
```
Comment (OussamaSaoudi-db) [inline CHANGELOG.md:16]:
```suggestion
7. `Expression::Unary`, `Expression::Binary`, and `Expression::Variadic` now wrap a struct of the
```

**Review by OussamaSaoudi-db — APPROVED** [review]
Besides the couple nits, LGTM 👍

File: CHANGELOG.md (hunk 3 above)
Comment (zachschuermann) [reply CHANGELOG.md:18]: did that one go in 0.5 tho?

Comment (OussamaSaoudi-db) [reply CHANGELOG.md]: Ah yeah good point

**Comment by roeap** [comment]
Since we are already introducing new error variants, should we see to get https://github.com/delta-io/delta-kernel-rs/pull/593 in before release?

**Comment by zachschuermann** [comment]
> Since we are already introducing new error variants, should we see to get #593 in before release?

considering nearly every release is a breaking release at this point i'm not too concerned

**Review by nicklan — APPROVED** [review]
nice thanks!

**Pull request merged**